src/Microsoft.ML.Data/DataView/TypedCursor.cs

```diff
@@ -217,6 +217,90 @@ public RowCursor<TRow>[] GetCursorSet(Func<int, bool> additionalColumnsPredicate
                  .ToArray();
         }
 
+        public static void ValidateMemberInfo(MemberInfo memberInfo, IDataView data)
+        {
+            if (!SchemaDefinition.CheckMemberInfo(memberInfo))
+                return;
+
+            var mappingNameAttr = memberInfo.GetCustomAttribute<ColumnNameAttribute>();
+            var singleName = mappingNameAttr?.Name ?? memberInfo.Name;
+
+            Type actualType = null;
+            bool isVector = false;
+            IEnumerable<Attribute> customAttributes = null;
+            switch (memberInfo)
+            {
+                case FieldInfo fieldInfo:
+                    InternalSchemaDefinition.GetMappedType(fieldInfo.FieldType, out actualType, out isVector);
+                    customAttributes = fieldInfo.GetCustomAttributes();
+                    break;
+
+                case PropertyInfo propertyInfo:
+                    InternalSchemaDefinition.GetMappedType(propertyInfo.PropertyType, out actualType, out isVector);
+                    customAttributes = propertyInfo.GetCustomAttributes();
+                    break;
+
+                default:
+                    Contracts.Assert(false);
+                    throw Contracts.ExceptNotSupp("Expected a FieldInfo or a PropertyInfo");
+            }
+
+            if (!actualType.TryGetDataKind(out _) && !DataViewTypeManager.Knows(actualType, customAttributes))
+            {
+                int colIndex;
+                data.Schema.TryGetColumnIndex(singleName, out colIndex);
+                DataViewType expectedType = data.Schema[colIndex].Type;
+                if (!actualType.Equals(expectedType.RawType))
+                    throw Contracts.ExceptParam(nameof(actualType), $"The expected type '{expectedType.RawType}' does not match the type of the '{singleName}' property: '{actualType.Name}'. Please change the {singleName} property to '{expectedType.RawType}'");
```

Comment:
```suggestion
                    throw Contracts.ExceptParam(nameof(actualType), $"The expected type '{expectedType.RawType}' does not match the type of the '{singleName}' member: '{actualType.Name}'. Please change the {singleName} member to '{expectedType.RawType}'");
```
The member could have been either a Property or a Field.